functions.sh

```diff
@@ -136,102 +138,104 @@ function get_config () {
-			IFS=' ' read -ra subdirs <<< "$(get_versions "${dir#./}")"
-			for subdir in "${subdirs[@]}"; do
-				versions+=( "$subdir" )
-			done
-		elif [ -a "$dir/Dockerfile" ]; then
-			versions+=( "${dir#./}" )
-		fi
-	done
-
-	if [ ${#versions[@]} -gt 0 ]; then
-		echo "${versions[@]%/}"
-	fi
+function get_versions() {
+  local prefix
+  prefix=${1:-.}
+  shift
+
+  local versions
+  local dirs=("$@")
+  if [ ${#dirs[@]} -eq 0 ]; then
+    IFS=' ' read -ra dirs <<<"$(echo "${prefix%/}/"*/)"
+  fi
+
+  for dir in "${dirs[@]}"; do
+    if [ -a "$dir/config" ]; then
+      local subdirs
+      IFS=' ' read -ra subdirs <<<"$(get_versions "${dir#./}")"
+      for subdir in "${subdirs[@]}"; do
+        versions+=("$subdir")
+      done
+    elif [ -a "$dir/Dockerfile" ]; then
+      versions+=("${dir#./}")
+    fi
+  done
+
+  if [ ${#versions[@]} -gt 0 ]; then
+    echo "${versions[@]%/}"
+  fi
 }
 
-function get_fork_name () {
-	local version
-	version=$1
-	shift
+function get_fork_name() {
+  local version
+  version=$1
+  shift
 
-	IFS='/' read -ra versionparts <<< "$version"
-	if [ ${#versionparts[@]} -gt 1 ]; then
-		echo "${versionparts[0]}"
-	fi
+  IFS='/' read -ra versionparts <<<"$version"
+  if [ ${#versionparts[@]} -gt 1 ]; then
+    echo "${versionparts[0]}"
+  fi
 }
 
-function get_full_version () {
-	local version
-	version=$1
-	shift
+function get_full_version() {
+  local version
+  version=$1
+  shift
 
-	grep -m1 'ENV NODE_VERSION ' "$version/Dockerfile" | cut -d' ' -f3
+  grep -m1 'ENV NODE_VERSION ' "$version/Dockerfile" | cut -d' ' -f3
 }
 
-function get_major_minor_version () {
-	local version
-	version=$1
-	shift
+function get_major_minor_version() {
+  local version
+  version=$1
+  shift
 
-	local fullversion
-	fullversion=$(get_full_version "$version")
+  local fullversion
+  fullversion=$(get_full_version "$version")
 
-	echo "$(echo "$fullversion" | cut -d'.' -f1).$(echo "$fullversion" | cut -d'.' -f2)"
+  echo "$(echo "$fullversion" | cut -d'.' -f1).$(echo "$fullversion" | cut -d'.' -f2)"
 }
 
-function get_tag () {
-	local version
-	version=$1
-	shift
-
-	local versiontype
-	versiontype=${1:-full}
-	shift
-
-	local tagversion
-	if [ "$versiontype" = full ]; then
-		tagversion=$(get_full_version "$version")
-	elif [ "$versiontype" = majorminor ]; then
-		tagversion=$(get_major_minor_version "$version")
-	fi
-
-	local tagparts
-	IFS=' ' read -ra tagparts <<< "$(get_fork_name "$version") $tagversion"
-	IFS='-'; echo "${tagparts[*]}"; unset IFS
+function get_tag() {
+  local version
+  version=$1
+  shift
+
+  local versiontype
+  versiontype=${1:-full}
+  shift
+
+  local tagversion
+  if [ "$versiontype" = full ]; then
+    tagversion=$(get_full_version "$version")
+  elif [ "$versiontype" = majorminor ]; then
+    tagversion=$(get_major_minor_version "$version")
+  fi
+
+  local tagparts
+  IFS=' ' read -ra tagparts <<<"$(get_fork_name "$version") $tagversion"
+  IFS='-'
+  echo "${tagparts[*]}"
+  unset IFS
```

Comment:
Why is this one multiline while the other isn't?

Reply:
(shfmt author, trying to clarify)

Multiple statements are always split in multiple lines - the first IFS line was a single statement, the second was three statements. Note the `;` characters.